nautobot/dcim/forms.py

```diff
@@ -2414,42 +2413,42 @@ class Meta:
             "mode",
             "untagged_vlan",
             "tagged_vlans",
+            "vrf",
         ]
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
-        # Limit LAG choices to interfaces which belong to the parent device (or VC master)
-        if "device" in self.initial:
-            device = Device.objects.filter(pk=self.initial["device"]).first()
+        if "pk" not in self.initial:
+            return
+
+        interfaces = (
+            Interface.objects.filter(pk__in=self.initial["pk"])
+            .select_related("device__location")
+            .only("id", "device", "device__location")
+        )
+        devices = Device.objects.filter(interfaces__in=interfaces).only("pk").distinct()
+        locations = Location.objects.without_tree_fields().order_by().filter(devices__in=devices).only("pk").distinct()
```

Comment:
Don't these also become a performance concern when bulk-editing a large set of interfaces?

Reply:
The Device query is a single join, the locations query is a single join with a subquery join. They both seem to be pretty fast for me and I've also constructed them to be as performant as possible.